mne/preprocessing/interpolate.py

```diff
@@ -111,35 +115,64 @@ def interpolate_bridged_electrodes(inst, bridged_idx):
     bads_orig = inst.info['bads']
     inst.info['bads'] = list()
 
+    # look for group of bad channels
+    nodes = sorted(set(chain(*bridged_idx)))
+    G_dense = np.zeros((len(nodes), len(nodes)))
+    # fill the edges with a weight of 1
+    for bridge in bridged_idx:
+        idx0 = nodes.index(bridge[0])
+        idx1 = nodes.index(bridge[1])
+        G_dense[idx0, idx1] = 1
+        G_dense[idx1, idx0] = 1
+    # look for connected components
+    _, labels = connected_components(csr_matrix(G_dense), directed=False)
```

Comment:
I would do this with https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.coo_matrix.html

Or if you know the matrices will always be small (they probably will) then what you have is fine. I wouldn't bother casting explicitly to `csr_matrix`, though

Reply:
It will always be small, a worst (impossible) case scenario would be a 256 electrodes cap with all electrodes bridged. So 256x256.

So `coo_matrix` to avoid the for loop and remove the `csr_matrix`. I'll check it later today.

Reply:
Up to you if you want to `coo_matrix` or not. It looks like you might have to loop to get `idx0` and `idx1`. In principle you could maybe use `np.searchsorted` to avoid even those loops but it starts to be less and less readable...

Reply:
I'll stick with the current version, I just changed the `.index` with a better `np.searchsorted`. I did not find an easy and readable way to convert the information to the format for `coo_matrix`. I was starting with:

```
nodes = sorted(set(chain(*bridged_idx)))
rows, cols = zip(*bridged_idx)
# need to convert the idx as with `nodes.index(idx)`
coo_matrix(np.ones(len(rows)), rows, cols, shape=(len(nodes), len(nodes)))
# ...
```

It doesn't look more readable, and the conversion of the `idx` requires the same loop.